### PR TITLE
feat(macro): throw useful error message if macro used without a plugin

### PR DESCRIPTION
**Pull request by thekip**

### CODE REVIEW

File: packages/macro/src/index.ts [hunk 1]
```diff
@@ -149,3 +149,29 @@ function getMacroType(tagName: string): string {
...
+  'plural',
+  'select',
+  'selectOrdinal',
+
+  // jsx
+  'Trans',
+  'Plural',
+  'Select',
+  'SelectOrdinal',
+];
```
Comment (Martin005) [inline packages/macro/src/index.ts:166]: I don't think it is a good idea to have two identical lists of tag names (the other is in the `getMacroType` function). Please merge these two concepts into a dictionary, update the `forEach` below, remove the `getMacroType` function and update getting the macro type on the line 43.

File: packages/macro/src/index.ts [hunk 2]
```diff
@@ -149,3 +149,29 @@ function getMacroType(tagName: string): string {
 }
 
 export default createMacro(macro)
```
Comment (Martin005) [inline packages/macro/src/index.ts:151]: I think that this export statement should be at the end of file.

File: packages/macro/src/index.ts [hunk 3]
```diff
@@ -149,3 +149,29 @@ function getMacroType(tagName: string): string {
...
+  'SelectOrdinal',
+];
+
+macros.forEach((name) => {
+  Object.defineProperty(module.exports, name, {
+    get() {
+      throw new Error(`The macro you imported from "@lingui/macro" is being executed outside the context of compilation with babel-plugin-macros. `
+        + `This indicates that you don't have the babel plugin "babel-plugin-macros" configured correctly. `
+        + `Please see the documentation for how to configure babel-plugin-macros properly: `
+        + 'https://github.com/kentcdodds/babel-plugin-macros/blob/master/other/docs/user.md');
```
Comment (Martin005) [inline packages/macro/src/index.ts:174]:
```suggestion
        + 'https://github.com/kentcdodds/babel-plugin-macros/blob/main/other/docs/user.md');
```

File: packages/macro/test/index.ts [hunk 4]
```diff
@@ -111,6 +120,17 @@ describe("macro", function () {
...
   })
 
+  it("Should throw error if used without babel-macro-plugin", async () => {
+    await expect(
+      async () => {
+        const mod = await import("../src/index");
+        return (mod as unknown as typeof import('@lingui/macro')).Trans
+      }
+    ).rejects.toThrowErrorMatchingInlineSnapshot(
+      `The macro you imported from "@lingui/macro" is being executed outside the context of compilation with babel-plugin-macros. This indicates that you don't have the babel plugin "babel-plugin-macros" configured correctly. Please see the documentation for how to configure babel-plugin-macros properly: https://github.com/kentcdodds/babel-plugin-macros/blob/master/other/docs/user.md`
```
Comment (Martin005) [inline packages/macro/test/index.ts:130]:
```suggestion
      `The macro you imported from "@lingui/macro" is being executed outside the context of compilation with babel-plugin-macros. This indicates that you don't have the babel plugin "babel-plugin-macros" configured correctly. Please see the documentation for how to configure babel-plugin-macros properly: https://github.com/kentcdodds/babel-plugin-macros/blob/main/other/docs/user.md`
```

**Review by Martin005 — CHANGES_REQUESTED** [review]

**Comment by thekip** [comment]
@Martin005 done

File: packages/macro/src/index.ts [hunk 5]
```diff
@@ -27,27 +27,41 @@ const getSymbolSource = (name: 'i18n' | 'Trans'): [source: string, identifier?:
...
-      throw nodes[0].buildCodeFrameError(`Unknown macro ${tagName}`)
-    }
 
-    if (macroType === "js") {
+    if (jsMacroTags.has(tagName)) {
       nodes.forEach((node) => {
         jsNodes.push(node.parentPath)
       })
-    } else {
+    } else if(jsxMacroTags.has(tagName)) {
```
Comment (Martin005) [inline packages/macro/src/index.ts:58]:
```suggestion
    } else if (jsxMacroTags.has(tagName)) {
```

**Review by Martin005 — APPROVED** [review]

**Pull request merged**